packages/server/graphql/mutations/removeMattermostAuth.ts

```diff
@@ -0,0 +1,41 @@
+import {GraphQLID, GraphQLNonNull} from 'graphql'
+import RemoveMattermostAuthPayload from '../types/RemoveMattermostAuthPayload'
+import {getUserId, isTeamMember} from '../../utils/authorization'
+import publish from '../../utils/publish'
+import standardError from '../../utils/standardError'
+import {SubscriptionChannel} from 'parabol-client/types/constEnums'
+import getMattermostAuthByTeamId from '../../postgres/queries/getMattermostAuthByTeamId'
+import removeMattermostAuth from '../../postgres/queries/removeMattermostAuth'
+
+export default {
+  name: 'RemoveMattermostAuth',
+  type: new GraphQLNonNull(RemoveMattermostAuthPayload),
+  description: 'Disconnect a team member from Slack',
+  args: {
+    teamId: {
+      type: new GraphQLNonNull(GraphQLID),
+      description: 'the teamId to disconnect from Mattermost'
+    }
+  },
+  resolve: async (_source, {teamId}, {authToken, socketId: mutatorId, dataLoader}) => {
+    const operationId = dataLoader.share()
+    const subOptions = {mutatorId, operationId}
+    const viewerId = getUserId(authToken)
+
+    // AUTH
+    if (!isTeamMember(authToken, teamId)) {
+      return standardError(new Error('Team not found'), {userId: viewerId})
```

Comment:
allowing any team member to remove any other team member's integration feels a little bad.

perhaps we have a 1:1 connection between team member & integration (just like with do with github & atlassian & slack)

then, when it comes time to send a message, we can pick the "best" token to use (ie the viewer's token, or any one else's, if a viewer token doesn't exist).